src/hotspot/share/utilities/vmError.hpp

```diff
@@ -154,22 +156,31 @@ class VMError : public AllStatic {
   static void print_vm_info(outputStream* st);
 
   // main error reporting function
+  [[noreturn]]
+  ATTRIBUTE_PRINTF(6, 7)
   static void report_and_die(Thread* thread, unsigned int sig, address pc, void* siginfo,
-                             void* context, const char* detail_fmt, ...) ATTRIBUTE_PRINTF(6, 7);
+                             void* context, const char* detail_fmt, ...);
 
+  [[noreturn]]
+  ATTRIBUTE_PRINTF(3, 0)
   static void report_and_die(int id, const char* message, const char* detail_fmt, va_list detail_args,
                              Thread* thread, address pc, void* siginfo, void* context,
-                             const char* filename, int lineno, size_t size) ATTRIBUTE_PRINTF(3, 0);
+                             const char* filename, int lineno, size_t size);
 
+  [[noreturn]]
   static void report_and_die(Thread* thread, unsigned int sig, address pc,
                              void* siginfo, void* context);
 
+  [[noreturn]]
+  ATTRIBUTE_PRINTF(6, 0)
   static void report_and_die(Thread* thread, void* context, const char* filename, int lineno, const char* message,
-                             const char* detail_fmt, va_list detail_args) ATTRIBUTE_PRINTF(6, 0);
+                             const char* detail_fmt, va_list detail_args);
 
+  [[noreturn]]
+  ATTRIBUTE_PRINTF(6, 0)
```

Comment:
I thought you were going to have to take BREAKPOINT out of vmassert in order to make report_and_die [[noreturn]] ?

Reply:
vmassert and friends are not noreturn yet, because of `Debugging`.  That's the next change in my local queue.